### PR TITLE
added a new data card selector widget for models

### PR DESCRIPTION
**Pull request by ARADDCC002**

### CODE REVIEW

File: frontend/src/MuiForms/DataCardSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  const handleInputChange = useCallback((_event: SyntheticEvent<Element, Event>, value: string) => {
+    setDataCardListQuery(value)
+  }, [])
+
+  const debounceOnInputChange = debounce((event: SyntheticEvent<Element, Event>, value: string) => {
+    handleInputChange(event, value)
+  }, 500)
+
+  if (isDataCardsError) {
+    if (isDataCardsError.status !== 413) {
```
Comment (ARADDCC012) [inline frontend/src/MuiForms/DataCardSelector.tsx:68]: Why are we not displaying an error in this case?

File: frontend/src/MuiForms/DataCardSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            setOpen(true)
+          }}
+          onClose={() => {
+            setOpen(false)
+          }}
+          disableClearable
+          isOptionEqualToValue={(option, value) => option.id === value.id}
+          getOptionLabel={(option) => option.name}
+          value={selectedDataCards || []}
+          filterOptions={(x) => x}
```
Comment (ARADDCC012) [inline frontend/src/MuiForms/DataCardSelector.tsx:93]: This isn't filtering anything, can be removed

File: frontend/src/MuiForms/DataCardSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      const updatedDataCards: ModelSearchResult[] = dataCards.filter((dataCard) => {
+        if (currentValue.includes(dataCard.id)) {
+          return dataCard
+        }
+      })
+      setSelectedDataCards(updatedDataCards)
+    }
+  }, [currentValue, dataCards])
+
+  const handleModelChange = useCallback(
```
Comment (ARADDCC012) [inline frontend/src/MuiForms/DataCardSelector.tsx:51]: Rename to something like `handleSelectedDataCardsChange`?

File: frontend/src/MuiForms/DataCardSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                fontStyle: 'italic',
+                color: theme.palette.customTextInput.main,
+              }}
+            >
+              Unanswered
+            </Typography>
+          )}
+          <Box sx={{ overflowX: 'auto', p: 1 }}>
+            <Stack spacing={1} direction='row'>
+              {currentValue.map((currentDataCard) => (
```
Comment (ARADDCC012) [inline frontend/src/MuiForms/DataCardSelector.tsx:138]: `currentDataCard` is the data card ID, but this isn't clear. Consider renaming to `currentDataCardId`

**Review by ARADDCC012 — CHANGES_REQUESTED** [review]

File: frontend/src/MuiForms/DataCardSelector.tsx (hunk 1 above)
Comment (ARADDCC002) [reply frontend/src/MuiForms/DataCardSelector.tsx:68]: This was left over from the entity selector sorry! Can be removed.

**Review by ARADDCC012 — APPROVED** [review]

**Pull request merged**